lightning/src/ln/chanmon_update_fail_tests.rs

```diff
@@ -2653,3 +2653,99 @@ fn test_permanent_error_during_handling_shutdown() {
 	check_closed_broadcast!(nodes[1], true);
 	check_added_monitors!(nodes[1], 2);
 }
+
+#[test]
+fn test_double_temp_error() {
+	// Test that it's OK to have multiple `ChainMonitor::update_channel` calls fail in a row.
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let mut nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	let (_, _, channel_id, _) = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known());
+
+	let (payment_preimage_1, _, _) = route_payment(&nodes[0], &[&nodes[1]], 1000000);
+	let (payment_preimage_2, _, _) = route_payment(&nodes[0], &[&nodes[1]], 1000000);
+
+	*nodes[1].chain_monitor.update_ret.lock().unwrap() = Some(Err(ChannelMonitorUpdateErr::TemporaryFailure));
+	// `claim_funds` results in a ChannelMonitorUpdate.
+	assert!(nodes[1].node.claim_funds(payment_preimage_1));
+	let (funding_tx, latest_update_1) = nodes[1].chain_monitor.latest_monitor_update_id.lock().unwrap().get(&channel_id).unwrap().clone();
+
+	*nodes[1].chain_monitor.update_ret.lock().unwrap() = Some(Err(ChannelMonitorUpdateErr::TemporaryFailure));
+	// Previously, this would've panicked due to a double-call to `Channel::monitor_update_failed`,
+	// which had some asserts that prevented it from being called twice.
+	assert!(nodes[1].node.claim_funds(payment_preimage_2));
+	*nodes[1].chain_monitor.update_ret.lock().unwrap() = Some(Ok(()));
+
+	let (_, latest_update_2) = nodes[1].chain_monitor.latest_monitor_update_id.lock().unwrap().get(&channel_id).unwrap().clone();
+	// TODO: should one of these `channel_monitor_updated` calls be omitted?
```

Comment:
The API allows either, I like the test of "make sure nothing happens if we provide a not-latest monitor update".